Spigot-Server-Patches/0594-Significantly-improve-performance-of-the-end-generat.patch

```diff
@@ -0,0 +1,58 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: SuperCoder7979 <25208576+SuperCoder7979@users.noreply.github.com>
+Date: Tue, 3 Nov 2020 23:48:05 -0600
+Subject: [PATCH] Significantly improve performance of the end generation
+
+This patch implements a noise cache for the end which significantly reduces the computation time of generation. This results in about a 3x improvement.
+
+Original code by SuperCoder7979 and Gegy in Lithium, licensed under LGPL-3.0 <Source: https://github.com/jellysquid3/lithium-fabric>
+
+Co-authored-by: Gegy <gegy1000@gmail.com>
+Co-authored-by: Dylan Xaldin <Puremin0rez515@gmail.com>
+Co-authored-by: pop4959 <pop4959@gmail.com>
+
+diff --git a/src/main/java/net/minecraft/server/WorldChunkManagerTheEnd.java b/src/main/java/net/minecraft/server/WorldChunkManagerTheEnd.java
+index 9c15cd303ccc2928be4e3e7080e74e8314d7e0f2..eb370c226d401ad0d1b29f054b34e6cfb5d192c1 100644
+--- a/src/main/java/net/minecraft/server/WorldChunkManagerTheEnd.java
++++ b/src/main/java/net/minecraft/server/WorldChunkManagerTheEnd.java
+@@ -3,6 +3,7 @@ package net.minecraft.server;
+ import com.google.common.collect.ImmutableList;
+ import com.mojang.serialization.Codec;
+ import com.mojang.serialization.codecs.RecordCodecBuilder;
++import it.unimi.dsi.fastutil.HashCommon; // Paper
+ import java.util.List;
+ 
+ public class WorldChunkManagerTheEnd extends WorldChunkManager {
+@@ -22,6 +23,8 @@ public class WorldChunkManagerTheEnd extends WorldChunkManager {
+     private final BiomeBase k;
+     private final BiomeBase l;
+     private final BiomeBase m;
++    private static final long[] keys = new long[8192]; // Paper
```

Comment:
This should probably be stored per-generator-instance. Even then there’s still a concern of handling threaded generation since this cache does not work concurrently- is it possible for concurrent generation of chunks on the same generator instance with Paper?